snapcraft/plugins/catkin.py

```diff
@@ -129,9 +129,8 @@ def _deps_from_packagesxml(self, f, pkg):
         try:
             tree = lxml.etree.parse(f)
         except lxml.etree.ParseError:
-            logger.warning('Unable to read "package.xml" file for "{}"'.format(
-                pkg))
-            return
+            raise EnvironmentError(
+                'Unable to read "package.xml" file for "{}"'.format(pkg))
```

Comment:
On Fri, Nov 6, 2015 at 12:55 PM, Leo Arias <notifications@github.com> wrote:

> In snapcraft/plugins/catkin.py
> <https://github.com/ubuntu-core/snapcraft/pull/83#discussion_r44151923>:
>
> > @@ -129,9 +129,8 @@ def _deps_from_packagesxml(self, f, pkg):
> >          try:
> >              tree = lxml.etree.parse(f)
> >          except lxml.etree.ParseError:
> > -            logger.warning('Unable to read "package.xml" file for "{}"'.format(
> > -                pkg))
> > -            return
> > +            raise EnvironmentError(
> > +                'Unable to read "package.xml" file for "{}"'.format(pkg))
>
> It would be good to include in this EnvironmentError the ParseError
> message.
>
> Feel free to ignore the following, because I'm not really sure about what
> I'm suggesting here:
> I'm not totally sure why you are changing lxml.etree.ParseError into
> EnvironmentError. I think I would just return the lxml exception, or wrap
> it in one of our own, like snapcraft.plugins.catking.CatkinError, or
> something.
>

Done, no try/catch there now.